### PR TITLE
Add support for function and mfa for tracing

### PR DESCRIPTION
**Pull request by ulissesalmeida**
As discussed on #70, it would be nice to give more flexbility
to control their tracing functions.

Here's a rough implementation of the idea. It add supports
for function and mfa to keep it consistent with transaction
and compensation.

### CODE REVIEW

File: lib/sage.ex [hunk 1]
```diff
@@ -279,26 +290,28 @@ defmodule Sage do
...
 
   Registering duplicated tracing callback is not allowed and would raise an
   `Sage.DuplicateTracerError` exception.
 
   All errors during execution of a tracing callbacks would be logged,
   but it won't affect Sage execution.
 
-  Tracing module must implement `Sage.Tracer` behaviour.
+  Tracer can be a module that must implement `Sage.Tracer` behaviour,
+  a function, or a tuple in a shape of `{module, function, [arguments]}`.
```
Comment (AndrewDryga) [inline lib/sage.ex]:
```suggestion
  a function, or a tuple in a shape of `{module, function, [extra_arguments]}`. 
  
  In any case, the function called should follow the definition of `c:Sage.Tracer.handle_event/3`
  and accept at least 3 required arguments that are documented by the callback.
```

Not sure if this is the best way to describe arguments of a function, but we definitely need to leave a note on it.

**Comment by AndrewDryga** [comment]
One little documentation thing and LGTM

File: lib/sage.ex (hunk 1 above)
Comment (ulissesalmeida) [reply lib/sage.ex]: Way better than I described ❤️

**Comment by AndrewDryga** [comment]
This looks great ❤️. Can you run the master branch for some time? I'll release the new version once you confirm it works good for you guys

**Comment by ulissesalmeida** [comment]
Sure, we'll do this week or next, and I'll let you know

**Comment by ulissesalmeida** [comment]
@AndrewDryga Just letting you know, we're using this change for since the PR creation, all good!

Thank you for accepting this change ❤️

**Pull request merged**